src/ray/raylet/lib/python/common_extension.cc

```diff
@@ -84,6 +84,34 @@ int PyObjectToUniqueID(PyObject *object, ObjectID *objectid) {
   }
 }
 
+int PyListStringToStringVector(PyObject *object,
+                               std::vector<std::string> *function_descriptor) {
+  if (function_descriptor == nullptr) {
+    PyErr_SetString(PyExc_TypeError, "function descriptor must be non-empty pointer");
+    return 0;
+  }
+  function_descriptor->clear();
+  std::vector<std::string> string_vector;
+  if (PyList_Check(object)) {
+    if (PyList_Size(object) == 0) {
+      return 1;
+    }
+    Py_ssize_t size = PyList_Size(object);
+    for (Py_ssize_t i = 0; i < size; ++i) {
+      PyObject *item = PyList_GetItem(object, i);
+      if (PyBytes_Check(item) == 0) {
+        PyErr_SetString(PyExc_TypeError, "PyListStringToStringVector gets bytes failed");
+        return 0;
+      }
+      function_descriptor->emplace_back(PyBytes_AsString(item), PyBytes_Size(item));
```

Comment:
There's a possibility that there could be null characters in the byte string, right? So maybe we should use a string constructor that takes a pointer and length.

Reply:
Actually, `vector<string>`'s emplace_back will call `string`'s constructor to create this string. `PyBytes_Size(item)` is also used, which guarantee that empty string will also work.